addons/sourcemod/scripting/vsh/base_boss.sp

```diff
@@ -426,3 +427,74 @@ public Action Timer_BossRageMusic(Handle hTimer, SaxtonHaleBase boss)
 	
 	return Plugin_Continue;
 }
+
+Action AttachEnt_SetTransmit(int iAttachEnt, int iClient)
+{
+	SetEdictFlags(iAttachEnt, GetEdictFlags(iAttachEnt) &~ FL_EDICT_ALWAYS);
+
+	int iOwner = GetEntPropEnt(iAttachEnt, Prop_Data, "m_pParent");
+	if (iOwner == INVALID_ENT_REFERENCE)
+		return Plugin_Stop;
+	
+	if (TF2_IsPlayerInCondition(iOwner, TFCond_Cloaked) || TF2_IsPlayerInCondition(iOwner, TFCond_Disguised) || TF2_IsPlayerInCondition(iOwner, TFCond_Stealthed))
+		return Plugin_Stop;
+	
+	if (iOwner != iClient || TF2_IsPlayerInCondition(iOwner, TFCond_Taunting))
+		return Plugin_Continue;
+	
+	return Plugin_Stop;
+}
+
+void ApplyBossEffects(SaxtonHaleBase boss)
+{
+	// TODO multiple effects
+	char sEffect1[64], sEffect2[64], sEffect3[64];
+	boss.CallFunction("GetParticleEffect", sizeof(sEffect1), sEffect1, sEffect2, sEffect3);
```

Comment:
May be better for `GetParticleEffect` to have 3 params as follows:
- int for the index to get, starting at 0
- char for the name of the particle
- int for the size of the char

It would keep calling the function while increasing index to go through all the particles until when there no more to get, with char param as empty.